examples/rtic_sai_wm8960.rs

```diff
@@ -0,0 +1,205 @@
+//! Audio playback using sai peripheral and imxrt-hal.
+//!
+//! Plays back a simple 440Hz (A note) simple square wave tone with the SAI peripheral
+//! to a Wolfson WM8960 codec on a number of the EVK boards.
+//!
+//! The audio stream itself is expected to be a 48000Hz 16bit stereo signal.
+
+#![no_main]
+#![no_std]
+
+/// Half of a sine wave (0 to pi)
+/// Can be used to generate a full sine wave by inverting (-1 * SIN_LUT[X]);
+const SIN_LUT: [u16; 256] = [
+    0, 402, 804, 1206, 1608, 2009, 2411, 2811, 3212, 3612, 4011, 4410, 4808, 5205, 5602, 5998,
+    6393, 6787, 7180, 7571, 7962, 8351, 8740, 9127, 9512, 9896, 10279, 10660, 11039, 11417, 11793,
+    12167, 12540, 12910, 13279, 13646, 14010, 14373, 14733, 15091, 15447, 15800, 16151, 16500,
+    16846, 17190, 17531, 17869, 18205, 18538, 18868, 19195, 19520, 19841, 20160, 20475, 20788,
+    21097, 21403, 21706, 22006, 22302, 22595, 22884, 23170, 23453, 23732, 24008, 24279, 24548,
+    24812, 25073, 25330, 25583, 25833, 26078, 26320, 26557, 26791, 27020, 27246, 27467, 27684,
+    27897, 28106, 28311, 28511, 28707, 28899, 29086, 29269, 29448, 29622, 29792, 29957, 30118,
+    30274, 30425, 30572, 30715, 30853, 30986, 31114, 31238, 31357, 31471, 31581, 31686, 31786,
+    31881, 31972, 32058, 32138, 32214, 32286, 32352, 32413, 32470, 32522, 32568, 32610, 32647,
+    32679, 32706, 32729, 32746, 32758, 32766, 32767, 32766, 32758, 32746, 32729, 32706, 32679,
+    32647, 32610, 32568, 32522, 32470, 32413, 32352, 32286, 32214, 32138, 32058, 31972, 31881,
+    31786, 31686, 31581, 31471, 31357, 31238, 31114, 30986, 30853, 30715, 30572, 30425, 30274,
+    30118, 29957, 29792, 29622, 29448, 29269, 29086, 28899, 28707, 28511, 28311, 28106, 27897,
+    27684, 27467, 27246, 27020, 26791, 26557, 26320, 26078, 25833, 25583, 25330, 25073, 24812,
+    24548, 24279, 24008, 23732, 23453, 23170, 22884, 22595, 22302, 22006, 21706, 21403, 21097,
+    20788, 20475, 20160, 19841, 19520, 19195, 18868, 18538, 18205, 17869, 17531, 17190, 16846,
+    16500, 16151, 15800, 15447, 15091, 14733, 14373, 14010, 13646, 13279, 12910, 12540, 12167,
+    11793, 11417, 11039, 10660, 10279, 9896, 9512, 9127, 8740, 8351, 7962, 7571, 7180, 6787, 6393,
+    5998, 5602, 5205, 4808, 4410, 4011, 3612, 3212, 2811, 2411, 2009, 1608, 1206, 804, 402,
+];
+
+/// Generate a sine wave sample
+fn sine(t: u32) -> u16 {
+    let p = t % 512;
+    let s = SIN_LUT[(p % 256) as usize];
+    if p < 256 {
+        (32768 + s) / 2
+    } else {
+        (32768 - s) / 2
+    }
+}
+
+/// Generate a square wave sample
+fn square(t: u32) -> u16 {
+    if (t % 128) > 64 {
+        32767
+    } else {
+        0
+    }
+}
+
+#[rtic::app(device = board, peripherals = false, dispatchers = [BOARD_SWTASK0])]
+mod app {
+
+    //
+    // Configure the demo below.
+    //
+
+    /// How frequently (milliseconds) should we poll audio
+    const AUDIO_POLL_MS: u32 = 1000 * (board::PIT_FREQUENCY / 1_000);
```

Comment:
It seems like the PIT is only used for printing debug information now. Variable/constant names and comments should probably be updated accordingly.